library/core/src/str/mod.rs

```diff
@@ -842,7 +842,9 @@ impl str {
     /// Lines are ended with either a newline (`\n`) or a carriage return with
     /// a line feed (`\r\n`).
     ///
-    /// The final line ending is optional.
+    /// The final line ending is optional. A string that ends with a final line
+    /// ending (carriage return or line feed) will return the same lines as an
+    /// otherwise identical string without a final line ending.
```

Comment:
> `(carriage return or line feed)`

Both a carriage return and a line feed together (in the right order: `\r\n`) also works. That it works for just a `\r` by itself without a following `\n` seems uninentional, so I guess that might be fixed at some point.

Maybe just leave this part out?

Reply:
Oops, I didn't mean to say "carriage return" there anyway, since that's wrong. I'll fix this.

Reply:
Pushed a change.